src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/scaleset/AzureScaleSetProvider.java

```diff
@@ -0,0 +1,382 @@
+                                                                                                                                              .withSku(selectScaleSetSku(instance.getHardware()
+                                                                                                                                                                                 .getType()))
+                                                                                                                                              .withExistingPrimaryNetworkSubnet(network,
+                                                                                                                                                                                azureSubnetName)
+                                                                                                                                              .withExistingPrimaryInternetFacingLoadBalancer(lb)
+                                                                                                                                              .withPrimaryInternetFacingLoadBalancerBackends(azureBackendPoolName)
+                                                                                                                                              .withPrimaryInternetFacingLoadBalancerInboundNatPools(azureNATPoolSSHName,
+                                                                                                                                                                                                    azureNATPoolPNPName,
+                                                                                                                                                                                                    azureNATPoolPNPSName)
+                                                                                                                                              .withoutPrimaryInternalLoadBalancer()
+                                                                                                                                              .withPopularLinuxImage(selectScaleSetLinuxImage(imageNameOrId))
+                                                                                                                                              .withRootUsername(vmAdminUsername);
+        VirtualMachineScaleSet.DefinitionStages.WithLinuxCreateManagedOrUnmanaged virtualMachineScaleSetStage2;
+
+        virtualMachineScaleSetStage2 = vmAdminSSHPubKey.map(virtualMachineScaleSetStage1::withSsh)
+                                                       .orElseGet(() -> virtualMachineScaleSetStage1.withRootPassword(vmAdminPassword));
+
+        VirtualMachineScaleSet virtualMachineScaleSet = virtualMachineScaleSetStage2
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "virtualMachineScaleSet". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1488)